team-builder/src/App.js

```diff
@@ -0,0 +1,37 @@
+import React, { useState } from 'react';
+import Team from "./components/Team";
+import TeamForm from "./components/TeamForm"
+import './App.css';
+
+function App() {
+  const [ teams, setTeams ] = useState([ {
+    id: 1,
+    name: "Richard Bishop",
+    email: "vimmer995@gmail.com",
+    role: "Evil genius/loving grandpa"
+  }
+]);
+
+// lets us create a new team (create function that contains--> takes same as team)
+  const addNewTeam = team => {
+    const newTeam = {
+      // id: is unique 
+      id: Date.now(),
+      name: team.name,
+      email: team.email,
+      role: team.role
+    };
+    setTeams([...teams, newTeam]);
+  }
+  // spread operator: check if can change with useParams
+
+  return (
+    <div className="App">
+       <h1>Team list</h1>
+      {/* // render props */}
+      <TeamForm addNewTeam = {addNewTeam} />
```

Comment:
Awesome job in passing down a function. This is a great uses case of passing a callback. Your formatting is off though there should not be spaces before or after the `=` so it should be
```jsx
addNewTeam={addNewTeam}
```

Reply:
ok i think that messes with my ocd lol i always feel that the equals sign should be seperated by spacing but i will change that habit

Reply:
In variables, yes.
```js
const variable = 0;
```

But with props since it is HTML there is no space so it should be
```html
<input type="text" value={variable} />
```

It is a difference between how `HTML` and `JavaScript` is formatted and when you are writing `JSX` you have to keep both in mind as you mix the styles.